tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -114,3 +116,22 @@ class TlcCounterexampleWriter(writer: PrintWriter) extends TlaCounterexampleWrit
   }
 }
 
+class JsonCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter {
+  override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState] ) : Unit = {
+    val json = new JsonWriter(writer)
+
+    val tlaStates = states.zipWithIndex.collect{
+      case ((tran,vars), i) =>
+        val state = vars.collect{
+          case (name,value) => OperEx(TlaOper.eq, NameEx(name), value)
+        }
+        TlaOperDecl(s"State${i+1}", List(), OperEx(TlaBoolOper.and, state.toList: _*))
+
+    }
+
+    val mod = new TlaModule(
+      "counterexample",
+      tlaStates ++ List(TlaOperDecl(s"InvariantViolation", List(), notInvariant)))
+    json.write(mod)
+  }
```

Comment:
I guess, the next logical step would be to extract the code that constructs `TlaModule`. Why shall we do that in three writers independently?

Reply:
right now, I suppose, this is not necessary: it will not bring any additional value, as I would have to add decomposition code from TLA into raw lists in the other two classes (Tla/TlcCounterexampleWriter). JsonCounterexampleWriter only needs to construct TLA, in order to use the already existing JsonWriter. This is not the case for the other two formats.